src/overlays/actors/ovl_En_Horse_Ganon/z_en_horse_ganon.c

```diff
@@ -25,29 +45,291 @@ const ActorInit En_Horse_Ganon_InitVars = {
+                             SkelAnime_GetFrameCount(&D_80A691B0[this->currentAnimation]->genericHeader), 2, 0.0f);
+    }
+}
+
+void func_80A68DB0(EnHorseGanon* this, GlobalContext* globalCtx) {
+    if (this->currentAnimation == 2) {
+        func_80A68870(this);
+    }
+
+    func_80A686A8(this, globalCtx);
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        func_80A68B20(this);
+    }
+}
+
+void func_80A68E14(EnHorseGanon* this, GlobalContext* globalCtx) {
+    u32 junk; // aligns stack properly
+    CollisionPoly* col;
+    f32 temp_ret;
+    Vec3f v;
+    s32 temp1;
+
+    v.x = Math_Sins(this->actor.shape.rot.y) * 30.0f + this->actor.posRot.pos.x;
+    v.y = this->actor.posRot.pos.y + 60.0f;
+    v.z = Math_Coss(this->actor.shape.rot.y) * 30.0f + this->actor.posRot.pos.z;
+
+    temp_ret = func_8003C940(&globalCtx->colCtx, &col, &temp1, &v);
+
+    this->unk_1f4 = temp_ret;
+    this->actor.shape.rot.x = D_80A692D0 * Math_atan2f(this->actor.posRot.pos.y - temp_ret, 30.0f);
+}
+
+#ifdef NON_MATCHING
+void EnHorseGanon_Update(EnHorseGanon* this, GlobalContext* globalCtx)
+// simply needs to store a0 in s0 a few instructions earlier
+{
+    u8 padding[8];
+
+    D_80A692C4[this->updateFnIndex](this, globalCtx);
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 20.0f, 55.0f, 100.0f, 29);
+    this->actor.posRot2.pos = this->actor.posRot.pos;
+    this->actor.posRot2.pos.y += 70.0f;
+    Collider_CylinderUpdate(&this->actor, &this->colliderCylinder);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderCylinder.base);
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Ganon/EnHorseGanon_Update.s")
+#endif
+
+void func_80A68FA8(EnHorseGanon* this, GlobalContext* globalCtx, ColliderJntSphItem* colliderSphereItem) {
+    Vec3f sp4C;
+    Vec3f sp40;
+    s32 index = 0;
+    s32 loops = 0;
+    ColliderJntSphItem* temp_v0;
+
+    if (this->colliderSphere.count > 0) {
+        do {
+            sp4C.x = this->colliderSphere.list[index].dim.modelSphere.center.x;
+            sp4C.y = this->colliderSphere.list[index].dim.modelSphere.center.y;
+            sp4C.z = this->colliderSphere.list[index].dim.modelSphere.center.z;
+
+            func_800A6408(colliderSphereItem, this->colliderSphere.list[index].dim.joint, &sp4C, &sp40);
+
+            this->colliderSphere.list[index].dim.worldSphere.center.x = (s16)sp40.x;
+            this->colliderSphere.list[index].dim.worldSphere.center.y = (s16)sp40.y;
+            this->colliderSphere.list[index].dim.worldSphere.center.z = (s16)sp40.z;
+
+            temp_v0 = &this->colliderSphere.list[index];
+            temp_v0->dim.worldSphere.radius = (s16)(temp_v0->dim.scale * (f32)temp_v0->dim.modelSphere.radius);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Ganon/func_80A68FA8.s")
+            index += 1;
+            loops += 1;
+        } while (loops < this->colliderSphere.count);
```

Comment:
Im not very inexperienced at loops so I cant be of much help, but afaik most do while's can be written as normal for or while loops, and that would be preferred. feel free to ask in the discord about it and someone who knows more can help